src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -392,39 +366,63 @@ protected override void OnScriptFileChanged(object sender, FileSystemEventArgs e
             // if the request includes a body, add it to the request object 
             if (request.Content != null && request.Content.Headers.ContentLength > 0)
             {
-                string body = request.Content.ReadAsStringAsync().Result;
-                rawBody = body;
                 MediaTypeHeaderValue contentType = request.Content.Headers.ContentType;
                 Dictionary<string, object> jsonObject;
-                if (contentType != null && contentType.MediaType == "application/json" &&
-                    TryDeserializeJson(body, out jsonObject))
+                object body = null;
+                if (contentType != null)
                 {
-                    // if the content - type of the request is json, deserialize into an object
-                    requestObject["body"] = jsonObject;
+                    if (contentType.MediaType == "application/json")
+                    {
+                        body = request.Content.ReadAsStringAsync().Result;
+                        if (TryDeserializeJson((string)body, out jsonObject))
+                        {
+                            // if the content - type of the request is json, deserialize into an object
+                            rawBody = (string)body;
+                            body = jsonObject;
+                        }
+                    }
+                    else if (contentType.MediaType == "application/octet-stream")
```

Comment:
Still thinking more on the right model for this. In the tests, I've added an http trigger test that specifies dataType binary so the script receives a byte[], however I'm wondering if I should key that off of the dataType only, or content type as I'm doing here.